File: src/core/cartesianPlane/translate/translate.ts
import { numberFns } from '../../numberFns';
import { cartesianPointType } from '../cartesianPoint';

type paramsType = {
    readonly point: cartesianPointType;
    readonly factor: cartesianPointType;
}

export function translate({ point, factor }: paramsType): cartesianPointType {
    return {
        x: numberFns.fix(point.x + factor.x),
        y: numberFns.fix(point.y + factor.y),
    };
} 

File: backend_java/src/main/java/com/AirplaneCoordinates/Features/Translate.java
package com.AirplaneCoordinates.Features;

import com.AirplaneCoordinates.Core.Plane.Cartesian.CartesianOperations;
import com.AirplaneCoordinates.Core.Plane.Cartesian.CartesianPoint;
import com.AirplaneCoordinates.Core.Plane.Conversion.PlaneConversion;
import com.AirplaneCoordinates.Core.Plane.Generic.PlanePoint;

public final class Translate {
    public static PlanePoint execute(final PlanePoint point, final CartesianPoint factor) {
        switch (point.planeType) {
            case CARTESIAN:
                return PlanePoint.fromCartesian(
                    CartesianOperations.sum(
                        point.toCartesian(),
                        factor
                    ).round()
                );
            case POLAR:
                return PlanePoint.fromPolar(
                    PlaneConversion.cartesianToPolar(
                        CartesianOperations.sum(
                            point.toCartesian(),
                            factor
                        )
                    ).round()
                );
            default:
                throw new RuntimeException("planeType is required");
        }
    }
}
